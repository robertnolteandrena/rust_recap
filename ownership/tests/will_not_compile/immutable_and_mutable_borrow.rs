fn main() {
    let mut a = String::from("Hello");
    let mutable_borrow = &mut a;
    let immutable_borrow = &a;
    println!("{} {}", mutable_borrow, immutable_borrow);
}
