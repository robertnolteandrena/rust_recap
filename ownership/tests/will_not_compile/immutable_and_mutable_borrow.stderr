error[E0502]: cannot borrow `a` as immutable because it is also borrowed as mutable
 --> tests/will_not_compile/immutable_and_mutable_borrow.rs:4:28
  |
3 |     let mutable_borrow = &mut a;
  |                          ------ mutable borrow occurs here
4 |     let immutable_borrow = &a;
  |                            ^^ immutable borrow occurs here
5 |     println!("{} {}", mutable_borrow, immutable_borrow);
  |                       -------------- mutable borrow later used here
